doc/run-cloud-build-locally.md

```diff
@@ -0,0 +1,197 @@
+
+SINK_SERVER_NAME=sink-server-jdbc SINK_SERVER_PORT=9002 docker compose -f docker/sink-compose.yml -p sink-server-jdbc up --force-recreate --remove-orphans -d
+```
+
+Wait for the servers to become ready and confirm via the command:
+
+```bash
+docker ps
+```
+
+You should get output similar to:
+
+```
+CONTAINER ID   IMAGE                     COMMAND                  CREATED              STATUS              PORTS                    NAMES
+3a23504297d0   hapiproject/hapi:latest   "java --class-path /…"   About a minute ago   Up About a minute   0.0.0.0:9002->8080/tcp   sink-server-jdbc
+c13ad0618e65   hapiproject/hapi:latest   "java --class-path /…"   2 minutes ago        Up 2 minutes        0.0.0.0:9001->8080/tcp   sink-server-search
+1f301480d87f   hapiproject/hapi:latest   "java --class-path /…"   15 minutes ago       Up 15 minutes       0.0.0.0:8091->8080/tcp   hapi-server
+815bc05a6005   postgres                  "docker-entrypoint.s…"   15 minutes ago       Up 15 minutes       0.0.0.0:5432->5432/tcp   hapi-fhir-db
+```
+
+### 2. Compile the Java modules
+
+```bash
+mvn --no-transfer-progress -e -T 2C install -Dlicense.skip=true -Dspotless.apply.skip=true
+```
+
+### 3. Upload sample synthetic data to HAPI stored in
+
+[sample_data](https://github.com/google/fhir-data-pipes/blob/master/synthea-hiv/sample_data)
+to the FHIR server that you brought up using the
+[Synthea data uploader](https://github.com/google/fhir-data-pipes/blob/master/synthea-hiv/README.md#Uploader).
+
+The uploader requires that you install the `uploader` module requirements first.
+You can do this by running:
+
+```bash
+pip3 install -r ./synthea-hiv/uploader/requirements.txt
+```
+
+Please note, it is a good idea to first create a Python `virtualenv` before
+running the above command to avoid conflicts with other Python packages you may
+have installed globally. You can do this by running:
+
+```bash
+$ virtualenv -p python3.8 venv
+$ . ./venv/bin/activate
+```
+
+If you had this already set up, just activate your existing virtualenv.
+
+```bash
+$ workon venv
+```
+
+Then, you can run the uploader script to upload the synthetic data to the FHIR
+server.
+
+For example, to upload to the HAPI FHIR server brought up in the previous step,
+run:
+
+```bash
+python3 ./synthea-hiv/uploader/main.py HAPI http://localhost:9001/fhir \
+--input_dir ./synthea-hiv/sample_data --cores 8
+```
+
+Depending on your machine, using too many cores may slow down your machine or
+cause JDBC connection pool errors with the HAPI FHIR server. Reducing the number
+of cores using the `--cores` flag should help at the cost of increasing the time
+to upload the data.
+
+### 4. Build the e2e controller/spark image
+
+```bash
+docker build -t ${_REPOSITORY}/e2e-tests/controller-spark:${_TAG} -f e2e-tests/controller-spark/Dockerfile .
+```
+
+### 5. Build and run the batch pipeline
+
+Use the Maven-built image (or rebuild with `pipelines/batch/docker` if you
```

Comment:
This is not very clear; what do you mean by "maven-built image"? Where did we create this docker image earlier?